go/enclave/gas/oracle.go

```diff
@@ -44,45 +67,92 @@ func (o *oracle) EstimateL1StorageGasCost(tx *types.Transaction, block *types.He
 	return o.calculateL1Cost(block, header, encodedTx)
 }
 
-func (o *oracle) EstimateL1CostForMsg(args *gethapi.TransactionArgs, block *types.Header, header *common.BatchHeader) (*big.Int, error) {
+func (o *oracle) EstimateL1CostForMsg(args *gethapi.TransactionArgs, header *common.BatchHeader) (*big.Int, error) {
 	encoded, err := rlp.EncodeToBytes(args)
 	if err != nil {
 		return nil, err
 	}
 
-	return o.calculateL1Cost(block, header, encoded)
+	return o.calculateL1Cost(nil, header, encoded)
+}
+
+func (o *oracle) SubmitL1Block(ctx context.Context, headBlock *types.Header) error {
+	o.headMutex.Lock()
+	defer o.headMutex.Unlock()
+	o.headBlock = headBlock
+	calculateBlobs := headBlock.ExcessBlobGas == nil
+
+	baseFeeSum := big.NewInt(0)
+	blobFeeSum := big.NewInt(0)
+	count := 0
+
+	b := headBlock
+	var err error
+	for ; count < MovingAverageWindow; count++ {
```

Comment:
is this run once to populate? seems expensive to get blocks in the history just for this rather than keep it in the oracle

Reply:
ah, ok